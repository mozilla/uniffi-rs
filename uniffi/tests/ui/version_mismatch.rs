// This should fail with a version mismatch.
uniffi::assert_compatible_version!("0.0.1"); // An error message would go here.

fn main() {}
