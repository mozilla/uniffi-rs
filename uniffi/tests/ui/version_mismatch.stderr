error[E0080]: evaluation of constant value failed
 --> tests/ui/version_mismatch.rs:2:1
  |
2 | uniffi::assert_compatible_version!("0.0.1"); // An error message would go here.
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ attempt to compute `0_usize - 1_usize`, which would overflow
  |
  = note: this error originates in the macro `uniffi::deps::static_assertions::const_assert` which comes from the expansion of the macro `uniffi::assert_compatible_version` (in Nightly builds, run with -Z macro-backtrace for more info)
